src/js/plugin/PluginLoader.js

```diff
@@ -0,0 +1,98 @@
+import config from "../config/config";
+
+import URLUtil from "../helpers/URLUtil";
+import PluginDispatcher from "./external/PluginDispatcher";
+import PluginEvents from "./external/PluginEvents";
+import PluginMountPoints from "./external/PluginMountPoints";
+import PluginActions from "./external/PluginActions";
+import PluginDispatcherProxy from "./PluginDispatcherProxy";
+
+const PLUGIN_STARTUP_TIMEOUT = 10000; // in ms
+
+const PluginLoader = {
+  load: function (pluginId, pluginURI) {
+    return new Promise(function (resolve, reject) {
+
+      // Create plugin host (hidden iframe)
+      const pluginHost = document.createElement("iframe");
+      pluginHost.setAttribute("scrolling","no");
+      pluginHost.setAttribute("width","1");
+      pluginHost.setAttribute("height","1");
+      pluginHost.setAttribute("frameborder","0");
+      pluginHost.setAttribute("style",
+        "position:absolute; top:-1px; left:-1px;");
+      pluginHost.setAttribute("sandbox","allow-same-origin allow-scripts");
+      pluginHost.setAttribute("src", URL.createObjectURL(new Blob([
+        `<html><head></head><body>${pluginId}</body></html>`
+      ], {type: "text/html"})));
+
+      function handlePluginHostLoad() {
+        pluginHost.removeEventListener("load", handlePluginHostLoad);
+        pluginHost.removeEventListener("error", handlePluginHostError);
+
+        const pluginWindow = pluginHost.contentWindow;
+        const pluginDocument = pluginWindow.document;
+        var pluginTimeout = null;
+
+        // Inject plugin interface
+        pluginWindow.marathonPluginInterface = Object.freeze({
```

Comment:
Regarding the naming I have no strong feelings. We can stick with your naming.
I've named it UI and API because it's basically an interface to the Marathon UI Application, less to the plugin.